src/Bicep.Core/TypeSystem/Providers/ResourceTypeProviderFactory.cs

```diff
@@ -27,41 +24,30 @@ public ResourceTypeProviderFactory(IFileSystem fileSystem)
             this.fileSystem = fileSystem;
         }
 
-        public ResultWithDiagnostic<IResourceTypeProvider> GetResourceTypeProvider(ResourceTypesProviderDescriptor providerDescriptor)
+        public ResultWithDiagnostic<IResourceTypeProvider> GetResourceTypeProvider(Uri typesTgzUri, bool useAzLoader)
         {
-            var key = new ResourceTypeLoaderKey(providerDescriptor.Name, providerDescriptor.Version);
+            var key = new ResourceTypeLoaderKey(typesTgzUri, useAzLoader);
+            // TODO invalidate this cache on module force restore
             return cachedResourceTypeLoaders.GetOrAdd(key, _ =>
             {
                 try
                 {
-                    return GetDynamicallyLoadedResourceTypesProvider(providerDescriptor);
+                    var typesLoader = OciTypeLoader.FromDisk(fileSystem, key.TypesTgzUri);
+                    if (key.UseAzLoader)
+                    {
+                        return new(new AzResourceTypeProvider(new AzResourceTypeLoader(typesLoader)));
```

Comment:
There's a lot of custom logic in this loader - it wasn't possible to fully abstract loading az types. Perhaps in the long term we could try and remove the need for this by moving more logic into the types contract.